### DISCUSSION TITLE
Avalonia Accelerate and Open Source project usage and licensing

### DISCUSSION THREAD

**Discussion by JosefNemec** [AvaloniaUI/Avalonia#17149] [General]
It's great to hear that there's large amount of effort being done to improve Avalonia via Accelerate project, but I have some questions about use of Accelerate components in open source projects.

I develop [OSS project](https://github.com/JosefNemec/Playnite) currently built with WPF that I'm in process of rewriting from scratch and I was initially redoing everything in Avalonia, but for various reasons I'm staying on WPF for now. One of the reasons being not so great developer experience, mainly in terms of user friendly designer support, which is why I was quite happy to see that this is being addressed via Accelerate somewhat.

I don't mind personally paying for these Accelerate components, but my project also has two separate groups of external contributors and I'm interested in knowing how it's going to work, license wise, if I decide to include Accelerate components in my OSS project. The groups are:

1) standard code contributions to the core app project
2) add-on developers making various plugins and UI themes

Lets say I decide to include video player Accelerate component in my project, will now every person contributing code to the project need to acquire Accelerate license since it's being licensed "per seat" and that developer is technically new "seat" on the project? What about theme developers that will make new themes for views that include video player component? Or plugin developers wanting to use video player in their plugins?

I kind of expect theme developers to be required to buy a license for visual designer, which is honestly not great since majority of add-on developers on my project are non-programmer hobbyists, but the main question is about components being built into the core project, like video player or web view.

Lastly, bit on a technical side, how do play for license verification to work with OSS projects? How are licenses going to be verified and issued? Would just any drive by developer be able to fork the project and build it with included Accelerate components or are there going to be some special checks and verifications needed?

**Comment by MikeCodesDotNET** [comment]
I'll try my best to answer below: 

**Licensing approach**
We will follow a model similar to that of other control vendors. Any UI controls part of Avalonia Accelerate will require a valid license.

**Visual Designer**
The visual designer tool will also require a license to use.

**Per-seat licensing**
Our per-seat model means that anyone who wants to use or consume any Avalonia Accelerate features must have a valid license. We aim to make the license as affordable as possible, as we're looking for a high number of low-paying customers rather than a low number of high-paying customers. 
 
For open-source projects, I'd recommend using open-source controls that align better with your project's licensing model if using paid controls isn't viable. This is typically how it's been done with other OSS projects using frameworks like WinForms and WPF. 

Regarding tools like the visual designer, OSS projects can still benefit from it as it generates XAML. This means projects can be shared in the same way as they are today without requiring licenses for all contributors. 

We understand the complexities this introduces for open-source projects, especially those with diverse contributor bases. We're committed to supporting the open-source community while balancing the sustainability of our pro offerings.

**Reply by danielmeza** [reply]
> I'll try my best to answer below:
> 
> **Licensing approach** We will follow a model similar to that of other control vendors. Any UI controls part of Avalonia Accelerate will require a valid license.
> 
> **Visual Designer** The visual designer tool will also require a license to use.
> 
> **Per-seat licensing** Our per-seat model means that anyone who wants to use or consume any Avalonia Accelerate features must have a valid license. We aim to make the license as affordable as possible, as we're looking for a high number of low-paying customers rather than a low number of high-paying customers.
> 
> For open-source projects, I'd recommend using open-source controls that align better with your project's licensing model if using paid controls isn't viable. This is typically how it's been done with other OSS projects using frameworks like WinForms and WPF.
> 
> Regarding tools like the visual designer, OSS projects can still benefit from it as it generates XAML. This means projects can be shared in the same way as they are today without requiring licenses for all contributors.
> 
> We understand the complexities this introduces for open-source projects, especially those with diverse contributor bases. We're committed to supporting the open-source community while balancing the sustainability of our pro offerings.

Hi Mike, thanks for the clarification, I wonder 
how  the recent changes in the license model affect this anounce, do FOSS mantainer need to keep using free UI libraries or are they able to use the new community license to acess controls like the TreeDataGrid? https://avaloniaui.net/blog/building-a-sustainable-future-for-avalonia

Im validating this cause we are getting several issues with the rendering performance in this app https://github.com/chkr1011/mqttMultimeter, I have made lots of imprevements in the data management side by including back pressure handling and ofloading the UI thread but virtualization is still a problem specially when there are more than 100K items to be render in the data grid.

**Reply by MikeCodesDotNET** [reply]
Hi @danielmeza, 

The Community edition of Accelerate is tooling only. You can read more about it [here](https://avaloniaui.net/accelerate/community).

**Comment by cirrusone** [comment]
I know there will likely be more forthcoming info but relating to this quote:
_Our per-seat model means that anyone who wants to use or consume any Avalonia Accelerate features must have a valid license._

Say I have 10 separate clients who I develop different commercial applications and all contain AA features such as web dialog. Each of those 10 clients sell the software to thousands of customers. If I am the only dev that builds the project is only 1 licence required? The terminology 'consume any Avalonia Accelerate features' is a little vague on who the consumer is, the dev or the app end users.

**Reply by MikeCodesDotNET** [reply]
I probably didn't word that particularly well. 

It's per developer using Avalonia Accelerate. In your case, you'd only require a single seat, regardless of how many clients or customers you have.

**Comment by xmaxrayx** [comment]
you can make them in dynamic library and just publish main project without exposing the paid/close-source stuff,

unless you use AGPL then u better find another lib that's opensource.

**Reply by MikeCodesDotNET** [reply]
If I've understood correctly what you're suggesting, this would breach the license terms. Every developer building a project that depends on Accelerate components must have a valid license. You cannot simply package the paid components into a dynamic library and ship it as a workaround to avoid properly licensing the components.

**Reply by xmaxrayx** [reply]
> If I've understood correctly what you're suggesting, this would breach the license terms. Every developer building a project that depends on Accelerate components must have a valid license. You cannot simply package the paid components into a dynamic library and ship it as a workaround to avoid properly licensing the components.

fair tbh i'm not interested with paid components so didn't read the license maybe just web app and hot reload thing but seems skipping linux isn't a problem to me ether and hot reload kind luxury so don't care.

even when you add a bunch of layers then load paid componentes without need them to run the app?
```txt
[main app]
[video service] , [other services]
[paid components] + [vlc player]
```
if yes ,then dotnot and anything be used as library like ffmpeg can't be used bc they don't have avalonia licenses or if it's 2-direct   work only?
what about add service/plugin with app api?
bc i know ppl do this or advice with problematic gbl / lgbl stuff unless its gbl3 or agbl.
